addons/mail/controllers/mail.py

```diff
@@ -219,17 +219,8 @@ def mail_thread_message_redirect(self, message_id, **kwargs):
             else:
                 raise Unauthorized()
 
-        url_params = {
-            'highlight_message_id': message_id,
-        }
         if message.model == 'discuss.channel':
-            url_params['active_id'] = message.res_id
-            url_params['action'] = 'mail.action_discuss'
+            url = f'/odoo/{message.res_id}/action-mail.action_discuss?highlight_message_id={message_id}'
```

Comment:
active_id in discuss is a little more special, they are alphanumeric, so they need to be in the queryString.

```suggestion
            url = f'/odoo/action-mail.action_discuss?active_id={message.res_id}&highlight_message_id={message_id}'
```